src/main/java/net/atos/zac/aanvraag/ProductaanvraagService.java

```diff
@@ -270,15 +275,20 @@ public ProductaanvraagDimpact getProductaanvraag(final ORObject productaanvraagO
         }
     }
 
-    private void addInitiator(final String bsn, final URI zaak, final URI zaaktype) {
+    private void addInitiator(final Betrokkene betrokkene, final URI zaak, final URI zaaktype) {
         final RolType initiator = ztcClientService.readRoltype(RolType.OmschrijvingGeneriekEnum.INITIATOR, zaaktype);
-        final RolNatuurlijkPersoon rolNatuurlijkPersoon = new RolNatuurlijkPersoon(
-                zaak,
-                initiator,
-                ROL_TOELICHTING,
-                new NatuurlijkPersoon(bsn)
-        );
-        zrcClientService.createRol(rolNatuurlijkPersoon);
+        final Rol<?> rol = getRol(betrokkene, zaak, initiator);
+        zrcClientService.createRol(rol);
+    }
+
+    private Rol<?> getRol(final Betrokkene betrokkene, final URI zaak, final RolType rolType) {
+        if (StringUtils.isNotBlank(betrokkene.getInpBsn()))
```

Comment:
Three if's make for a switch statement I think? Also easier to read.

Reply:
What would that look like, a switch statement over three different values?

Reply:
Done :-)